**Core -> Full Integration** [Build 15167](https://ci.wildfly.org/buildConfiguration/WildFlyCore_PullRequest_GalleonWildFlyCoreFullIntegrationLinuxJdk21/567062) outcome was **FAILURE** using a merge of 8547d4b97282d58f75f722a87be06b8896c1af4f
Summary: Tests failed: 1 (1 new), passed: 4641, ignored: 66 Build time: 03:15:31
### Failed tests
```
org.wildfly.test.integration.microprofile.reactive.messaging.otel.AmqpReactiveMessagingAndOtelTestCase(layers-test).testOpenTelemetryTracingAlwaysDisabledAtChannel: java.lang.AssertionError: Unable to find 2 times [SpanWithExpectedCount[span=POST /mp-rm-amqp-otel/, expectedCount=1]SpanWithExpectedCount[span=testamqp publish, expectedCount=1]SpanWithExpectedCount[span=testamqp receive, expectedCount=1]SpanWithExpectedCount[span=disabled-tracing publish, expectedCount=1]SpanWithExpectedCount[span=disabled-tracing receive, expectedCount=1]]
	at org.wildfly.test.integration.microprofile.reactive.messaging.otel.BaseReactiveMessagingAndOtelTest.checkJaegerTraces(BaseReactiveMessagingAndOtelTest.java:467)
	at org.wildfly.test.integration.microprofile.reactive.messaging.otel.BaseReactiveMessagingAndOtelTest.testOpenTelemetryTraces(BaseReactiveMessagingAndOtelTest.java:172)
	at org.wildfly.test.integration.microprofile.reactive.messaging.otel.BaseReactiveMessagingAndOtelTest.testOpenTelemetryTracingAlwaysDisabledAtChannel(BaseReactiveMessagingAndOtelTest.java:350)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
------- Stdout: -------
=============================
TestName: testOpenTelemetryTracingAlwaysDisabledAtChannel
=============================
 [0m02:58:27,137 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-3) WFLYUT0022: Unregistered web context: '/lra-coordinator' from server 'default-server'
 [0m [0m02:58:27,138 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0022: Unregistered web context: '/lra-participant-narayana-proxy' from server 'default-server'
 [0m [0m02:58:27,139 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-7) WFLYJCA0010: Unbound data source [java:jboss/datasources/ExampleDS]
 [0m [0m02:58:27,140 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 36) WFLYUT0022: Unregistered web context: '/mp-rm-amqp-otel' from server 'default-server'
 [0m [0m02:58:27,141 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-5) WFLYJCA0019: Stopped Driver service with driver-name = h2
 [0m [0m02:58:27,142 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0019: Host default-host stopping
 [0m [0m02:58:27,146 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-7) WFLYUT0008: Undertow HTTP listener default suspending
 [0m [0m02:58:27,157 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-7) WFLYUT0007: Undertow HTTP listener default stopped, was bound to [::1]:8080
 [0m [0m02:58:27,158 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-7) WFLYUT0004: Undertow 2.4.1.Final stopping
 [0m [0m02:58:27,187 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-5) WFLYSRV0028: Stopped deployment mp-rm-amqp-otel.war (runtime-name: mp-rm-amqp-otel.war) in 50ms
 [0m [0m02:58:27,200 INFO  [org.jboss.as] (MSC service thread 1-4) WFLYSRV0050: WildFly 41.0.0.Beta1-SNAPSHOT (WildFly Core 33.0.0.Beta3-SNAPSHOT) stopped in 64ms
 [0m [0m02:58:27,202 INFO  [org.jboss.as] (MSC service thread 1-4) WFLYSRV0049: WildFly 41.0.0.Beta1-SNAPSHOT (WildFly Core 33.0.0.Beta3-SNAPSHOT) starting
 [0m [0m02:58:27,339 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0039: Creating http management service using socket-binding (management-http)
 [0m [0m02:58:27,353 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 45) WFLYJCA0004: Deploying JDBC-compliant driver class org.h2.Driver (version 2.4)
 [0m [33m02:58:27,378 WARN  [org.wildfly.extension.elytron] (MSC service thread 1-9) WFLYELY00023: KeyStore file '/opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/standalone/configuration/application.keystore' does not exist. Used blank.
 [0m [33m02:58:27,379 WARN  [org.wildfly.extension.elytron] (MSC service thread 1-8) WFLYELY01084: KeyStore /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/standalone/configuration/application.keystore not found, it will be auto-generated on first use with a self-signed certificate for host localhost
 [0m [0m02:58:27,382 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 54) WFLYCLINF0001: Activating Infinispan subsystem.
 [0m [0m02:58:27,401 INFO  [org.wildfly.extension.health] (ServerService Thread Pool -- 53) WFLYHEALTH0001: Activating Base Health Subsystem
 [0m [0m02:58:27,401 INFO  [org.jboss.as.jaxrs] (ServerService Thread Pool -- 56) WFLYRS0016: RESTEasy version 7.0.2.Final
 [0m [0m02:58:27,407 INFO  [org.wildfly.extension.io] (ServerService Thread Pool -- 55) WFLYIO001: Worker 'default' has auto-configured to 8 IO threads with 64 max task threads based on your 4 available processors
 [0m [0m02:58:27,410 INFO  [org.jboss.as.connector] (MSC service thread 1-3) WFLYJCA0009: Starting Jakarta Connectors Subsystem (WildFly/IronJacamar 3.0.19.Final)
 [0m [0m02:58:27,411 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-7) WFLYJCA0018: Started Driver service with driver-name = h2
 [0m [0m02:58:27,415 INFO  [org.wildfly.extension.micrometer] (ServerService Thread Pool -- 62) WFLYMMTREXT0012: Additional metrics systems discovered while configuring Micrometer: OpenTelemetry Metrics. Please refer to the documentation for more information.
 [0m [0m02:58:27,418 INFO  [org.wildfly.extension.microprofile.config.smallrye] (ServerService Thread Pool -- 63) WFLYCONF0001: Activating MicroProfile Config Subsystem
 [0m [0m02:58:27,418 INFO  [org.wildfly.extension.microprofile.faulttolerance] (ServerService Thread Pool -- 64) WFLYMPFTEXT0001: Activating MicroProfile Fault Tolerance subsystem.
 [0m [0m02:58:27,419 INFO  [org.wildfly.extension.microprofile.jwt.smallrye] (ServerService Thread Pool -- 66) WFLYJWT0001: Activating MicroProfile JWT Subsystem
 [0m [0m02:58:27,419 INFO  [org.wildfly.extension.microprofile.health.smallrye] (ServerService Thread Pool -- 65) WFLYMPHEALTH0001: Activating MicroProfile Health Subsystem
 [0m [0m02:58:27,420 INFO  [org.jboss.as.ejb3] (MSC service thread 1-7) WFLYEJB0481: Strict pool slsb-strict-max-pool is using a max instance size of 64 (per class), which is derived from thread worker pool sizing.
 [0m [0m02:58:27,421 INFO  [org.wildfly.extension.microprofile.lra.participant] (ServerService Thread Pool -- 68) WFLYTXLRAPARTICIPANT0001: Activating MicroProfile LRA Participant Subsystem with system property (lra.coordinator.url) value as http://localhost:8080/lra-coordinator/lra-coordinator
 [0m [0m02:58:27,421 INFO  [org.wildfly.extension.microprofile.lra.coordinator] (ServerService Thread Pool -- 67) WFLYTXLRACOORD0001: Activating MicroProfile LRA Coordinator Subsystem
 [0m [0m02:58:27,420 INFO  [org.jboss.as.ejb3] (MSC service thread 1-8) WFLYEJB0482: Strict pool mdb-strict-max-pool is using a max instance size of 16 (per class), which is derived from the number of CPUs on this host.


```

--- a/server/src/main/java/org/jboss/as/server/controller/resources/ModuleInfoHandler.java
+++ b/server/src/main/java/org/jboss/as/server/controller/resources/ModuleInfoHandler.java
@@ -5,8 +5,6 @@
 
 package org.jboss.as.server.controller.resources;
 
-import java.lang.reflect.Field;
-
 import org.jboss.as.controller.AttributeDefinition;
 import org.jboss.as.controller.ObjectListAttributeDefinition;
 import org.jboss.as.controller.ObjectTypeAttributeDefinition;
@@ -106,19 +104,15 @@ public class ModuleInfoHandler implements OperationStepHandler {
         String id = JBossModulesNameUtil.canonicalModuleIdentifier(moduleName, slot);
         ModuleLoader loader = Module.getBootModuleLoader();
         try {
-            ModuleLoaderMXBean mxBean = getMxBean(loader);
+            loader.loadModule(id);
+            ModuleLoaderMXBean mxBean = ModuleLoadingResourceDefinition.getMxBean(id);
             ModuleInfo moduleInfo = mxBean.getModuleDescription(id);
             context.getResult().set(populateModuleInfo(moduleInfo));
+        } catch(OperationFailedException e) {
+            throw e;
         } catch (Exception e) {
             throw ServerLogger.ROOT_LOGGER.couldNotGetModuleInfo(id, e);
         }
-    }
-
-
-    protected ModuleLoaderMXBean getMxBean(ModuleLoader loader) throws ReflectiveOperationException {
-        Field mxBeanField = ModuleLoader.class.getDeclaredField("mxBean");
-        mxBeanField.setAccessible(true);
-        return (ModuleLoaderMXBean) mxBeanField.get(loader);
     }
 
     /*

--- a/server/src/main/java/org/jboss/as/server/controller/resources/ModuleLoadingResourceDefinition.java
+++ b/server/src/main/java/org/jboss/as/server/controller/resources/ModuleLoadingResourceDefinition.java
@@ -11,15 +11,23 @@ import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.MOD
 
 import java.io.File;
 import java.io.IOException;
+import java.lang.management.ManagementFactory;
 import java.lang.reflect.Field;
 import java.net.JarURLConnection;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.file.Path;
 import java.security.AccessController;
 import java.security.PrivilegedActionException;
 import java.security.PrivilegedExceptionAction;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Set;
+
+import javax.management.JMX;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
 
 import org.jboss.as.controller.AbstractRuntimeOnlyHandler;
 import org.jboss.as.controller.AttributeDefinition;
@@ -37,6 +45,7 @@ import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraint
 import org.jboss.as.controller.client.helpers.JBossModulesNameUtil;
 import org.jboss.as.controller.registry.ManagementResourceRegistration;
 import org.jboss.as.server.controller.descriptions.ServerDescriptions;
+import org.jboss.as.server.logging.ServerLogger;
 import org.jboss.dmr.ModelNode;
 import org.jboss.dmr.ModelType;
 import org.jboss.modules.LocalModuleFinder;
@@ -89,12 +98,28 @@ public class ModuleLoadingResourceDefinition extends SimpleResourceDefinition {
                 .setRuntimeOnly()
                 .setReplyType(ModelType.LIST)
                 .setReplyValueType(ModelType.STRING)
-                .setDeprecated(ModelVersion.create(1, 4, 0))
                 .setReadOnly()
                 .build();
 
          resourceRegistration.registerOperationHandler(definition, new ModuleLocationHandler());
          resourceRegistration.registerOperationHandler(ModuleInfoHandler.DEFINITION, ModuleInfoHandler.INSTANCE);
+    }
+
+    static ModuleLoaderMXBean getMxBean(String moduleName) throws OperationFailedException {
+        try {
+            MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+            ObjectName pattern = new ObjectName("jboss.modules:type=ModuleLoader,*");
+            Set<ObjectName> names = mbs.queryNames(pattern, null);
+            for (ObjectName name : names) {
+                ModuleLoaderMXBean moduleLoaderMXBean = JMX.newMXBeanProxy(mbs, name, ModuleLoaderMXBean.class);
+                if (moduleLoaderMXBean.queryLoadedModuleNames().contains(moduleName)) {
+                    return moduleLoaderMXBean;
+                }
+            }
+        } catch (MalformedObjectNameException e) {
+            throw ServerLogger.ROOT_LOGGER.couldNotGetModuleInfo(moduleName, e);
+        }
+        throw ServerLogger.ROOT_LOGGER.couldNotGetModuleInfo(moduleName, null);
     }
 
     /** Read attribute handler for "module-roots" */
@@ -204,28 +229,21 @@ public class ModuleLoadingResourceDefinition extends SimpleResourceDefinition {
     }
 
 
-    private static List<String> findResourcePaths(String moduleName) throws ModuleLoadException, ReflectiveOperationException, IOException, URISyntaxException {
+    private static List<String> findResourcePaths(String moduleName) throws ModuleLoadException, IOException, OperationFailedException, URISyntaxException {
         ModuleLoader moduleLoader = Module.getCallerModuleLoader();
-        ModuleLoaderMXBean loader = ModuleInfoHandler.INSTANCE.getMxBean(moduleLoader);
         moduleLoader.loadModule(JBossModulesNameUtil.parseCanonicalModuleIdentifier(moduleName));
+        ModuleLoaderMXBean loader = getMxBean(moduleName);
 
         List<String> result = new LinkedList<>();
-        for (ResourceLoaderInfo rl : loader.getResourceLoaders(moduleName)){
+        for (ResourceLoaderInfo rl : loader.getResourceLoaders(moduleName)) {
             final String location = rl.getLocation();
             if (location != null && !location.equals("null")) {
                 URL url = new URL(location);
-
-                switch (url.getProtocol()){
-
-                    case "jar": {
-                        JarURLConnection jarConnection = (JarURLConnection)url.openConnection();
-                        result.add(jarConnection.getJarFile().getName());
-
-                        break;
-                    }
-                    default: {
-                        result.add(new File(url.getFile() ).toString());
-                    }
+                if (url.getProtocol().equals("jar")) {
+                    JarURLConnection jarConnection = (JarURLConnection) url.openConnection();
+                    result.add(jarConnection.getJarFile().getName());
+                } else {
+                    result.add(Path.of(url.toURI()).toString());
                 }
             }
         }

--- a/server/src/main/resources/org/jboss/as/server/controller/descriptions/LocalDescriptions.properties
+++ b/server/src/main/resources/org/jboss/as/server/controller/descriptions/LocalDescriptions.properties
@@ -183,7 +183,6 @@ core.module-loading.module-roots.module-root=Filesystem location under which the
 core.module-loading.list-resource-loader-paths=Provides a list of filesystem paths of the module, if there are any and the module loader exposes them.
 core.module-loading.list-resource-loader-paths.module=The identifier of the module, in <name>[:<slot>] format. If the slot is not provided, a value of "main" is assumed.
 core.module-loading.list-resource-loader-paths.reply=A list, each of whose elements is a filesystem location from which the module can load resources.
-core.module-loading.list-resource-loader-paths.deprecated=Experimental and unsupported.
 core.module-loading.module-info=Displays information about modules in running system
 core.module-loading.module-info.name=Module name
 core.module-loading.module-info.slot=Module slot

